src/modules/category/category.service.ts

```diff
@@ -23,22 +27,30 @@ export class CategoryService {
   }
 
   async createCategory(data: CreateCategoryDto): Promise<Category> {
-    const existCategory = await this.categoryRepository.findOne({ name: data.name });
+    const existCategory = await this.categoryRepository.findOne({ title: data.title });
     if (existCategory) throw new MessageCodeError('category:exist');
     const newCategory = this.categoryRepository.create(data);
-    await this.categoryRepository.save(data);
-    return newCategory;
+    newCategory.seoId = slugify(data.title);
+    return this.categoryRepository.save(newCategory);
   }
 
   async updateCategory(data: UpdateCategoryDto): Promise<Category> {
     const category = await this.categoryRepository.findOne(data.id);
     if (!category) throw new MessageCodeError('category:notFound');
+    if (data.title) {
+      category.seoId = slugify(data.title);
+    }
```

Comment:
and if such a seoid already exists? when creating, you check, but not here

Reply:
done

Reply:
done